src/main/java/core/basesyntax/impl/StorageImpl.java

```diff
@@ -3,17 +3,60 @@
 import core.basesyntax.Storage;
 
 public class StorageImpl<K, V> implements Storage<K, V> {
+    private static final int STORAGE_SIZE = 10;
+    private K key;
+    private V value;
+    private int count = 0;
+    private K[] keys;
+    private V[] values;
+
+    public StorageImpl() {
+        this.keys = (K[]) new Object[STORAGE_SIZE];
+        this.values = (V[]) new Object[STORAGE_SIZE];
+    }
+
     @Override
     public void put(K key, V value) {
+        if (indexOfKey(keys, key) == -1 || key == null && get(key) != null) {
```

Comment:
after you fix previous comment, you will be able to remove this part:
```suggestion
        if (indexOfKey(keys, key) == -1) {
```